src/pybind/mgr/dashboard/tox.ini

```diff
@@ -1,30 +1,74 @@
 [tox]
-envlist = py27-{cov,lint,run,check},py3-{cov,lint,run,check}
+envlist =
+    py{27,3},
+    lint,
+    check,
+    run,
 skipsdist = true
-toxworkdir = {env:CEPH_BUILD_DIR}/dashboard
-minversion = 2.8.1
+minversion = 2.9.1
+
+[pytest]
+addopts =
+    --cov --cov-append --cov-report=term
+    --doctest-modules
+    --ignore=frontend/ --ignore=module.py
+    --instafail
 
 [testenv]
-setenv=
+deps =
+    -rrequirements.txt
```

Comment:
Just note, we removed those from `deps =` as a workaround for https://github.com/tox-dev/tox/issues/149 . This change will re-introduce this bug again?

(Well, I'm :+1: for changing this back)

Reply:
Wan't aware of that... As I understand, CI builds won't suffer from this as they will create virtualenvs from scratch. For non-CI builds, we still have the `--recreate` flag to force virtualenv creation.

Reply:
Hm, forcefully recreating the tox envs make things slower for little gain.

Reply:
Well, venvs would only need to be recreated in case of changing requirements, which is something happening quite sporadically. Besides, the previous approach wouldn't solve all issues either (e.g.: removed packages would remain there). Rather than going back to that workaround, why not exploring alternatives (pipfile) or tox plugins handling this situation?